lib/geometry/polygon.nit

```diff
@@ -0,0 +1,428 @@
+	# poly2.anti_clock_sort
+	# assert poly.intersects(poly2)
+	# ~~~
+	fun intersects(other: ConvexPolygon): Bool do
+		assert is_convex
+
+		var axes1 = get_axes
+		var axes2 = other.get_axes
+		for i in [0..axes1.length[ do
+			var axis = axes1[i]
+			var project1 = project(axis)
+			var project2 = other.project(axis)
+			if not project1.overlap(project2) then return false
+		end
+		for i in [0..axes2.length[ do
+			var axis = axes2[i]
+			var project1 = project(axis)
+			var project2 = other.project(axis)
+			if not project1.overlap(project2) then return false
+		end
+		return true
+	end
+
+	# Projection of the axis of the polygon for the intersection test
+	private fun project(axis: Point[Float]): Projection do
+		var min = axis.x * points[0].x + axis.y * points[0].y
+		var max = min
+		for i in [0..points.length[ do
+			var p = axis.x * points[i].x + axis.y * points[i].y
+			if p < min then min = p
+			if p > max then max = p
+		end
+		var projection = new Projection(min, max)
+		return projection
+	end
+
+	# Check if this polygon is convex
+	# ~~~
+	# var p1 = new Point[Float](0.0, 0.0)
+	# var p2 = new Point[Float](5.0, 0.0)
+	# var p3 = new Point[Float](0.0, 5.0)
+	# var p4 = new Point[Float](5.0, 5.0)
+	# var arr = new Array[Point[Float]].with_items(p1, p2, p3, p4)
+	# var poly = new ConvexPolygon.with_vertices(arr)
+	# poly.anti_clock_sort
+	# assert poly.is_convex
+	# ~~~
+	fun is_convex: Bool do
+		var prev = points[points.length - 2]
+		var curr = points[points.length - 1]
+		var next = points[0]
+		var is_ccw = turn_left(prev, curr, next)
+		for i in [1..points.length[ do
+			prev = curr
+			curr= next
+			next = points[i]
+			if turn_left(prev ,curr, next) != is_ccw then return false
+		end
+		return true
+	end
+
+	# Check if `p` is in the polygon
+	# ~~~
+	# var p1 = new Point[Float](0.0, 0.0)
+	# var p2 = new Point[Float](5.0, 0.0)
+	# var p3 = new Point[Float](0.0, 5.0)
+	# var p4 = new Point[Float](5.0, 5.0)
+	# var p5 = new Point[Float](2.5, 2.5)
+	# var arr = new Array[Point[Float]].with_items(p1, p2, p3, p4)
+	# var poly = new ConvexPolygon.with_vertices(arr)
+	# poly.anti_clock_sort
+	# assert poly.contain(p5)
+	# ~~~
+	fun contain(p : Point[Float]): Bool do
+		var prev = points[points.length - 1]
+		var curr = p
+		var next = points[0]
+		var is_ccw = turn_left(prev, curr, next)
+		for i in [1..points.length[ do
+			prev = next
+			next = points[i]
+			if turn_left(prev, curr, next) != is_ccw then return false
+		end
+		print points.length
+		return true
+	end
+
+	# Check if the order of the points in the polygon is counter-clockwise
+	# The vertices in the polygon need to be sorted
+	# ~~~
+	# var p1 = new Point[Float](0.0, 0.0)
+	# var p2 = new Point[Float](5.0, 0.0)
+	# var p3 = new Point[Float](0.0, 5.0)
+	# var p4 = new Point[Float](5.0, 5.0)
+	# var arr = new Array[Point[Float]].with_items(p1, p2, p3, p4)
+	# var poly = new ConvexPolygon.with_vertices(arr)
+	# poly.anti_clock_sort
+	# assert poly.is_ccw
```

Comment:
`is_ccw` is just an indication to know the order of the vertices in the polygon, it's used in the other algorithms. It supposes that the polygon's vertices are already sorted.